examples/example-web-app/components/SendV0TransactionButton.tsx

```diff
@@ -0,0 +1,187 @@
+import InfoIcon from '@mui/icons-material/Info';
+import {
+    Button,
+    ButtonGroup,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    Typography,
+} from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { useConnection, useWallet } from '@solana/wallet-adapter-react';
+import type { TransactionSignature } from '@solana/web3.js';
+import { PublicKey, TransactionMessage, VersionedTransaction } from '@solana/web3.js';
+import { useSnackbar } from 'notistack';
+import React, { useState } from 'react';
+
+import useGuardedCallback from '../utils/useGuardedCallback';
+
+type Props = Readonly<{
+    children?: React.ReactNode;
+    message: string;
+}>;
+
+function getBase64StringFromByteArray(byteArray: Uint8Array): string {
+    return btoa(String.fromCharCode.call(null, ...byteArray));
+}
+
+export default function SendV0TransactionButton({ children, message }: Props) {
+    const { enqueueSnackbar } = useSnackbar();
+    const { connection } = useConnection();
+    const { publicKey, sendTransaction, wallet } = useWallet();
+    const supportedTransactionVersions = wallet?.adapter.supportedTransactionVersions;
+    const [sendV0TransactionTutorialOpen, setSendV0TransactionTutorialOpen] = useState(false);
+    const [awaitingConfirmation, setAwaitingConfirmation] = useState(false);
+    const [previewSignature, setPreviewSignature] = useState<TransactionSignature | undefined>(undefined);
+    // const sendTransactionGuarded = useGuardedCallback(
+    //     async (transaction, connection) => {
+    //         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    //         return await sendTransaction!(transaction, connection);
+    //     },
+    //     [sendTransaction],
+    // );
+    return (
+        <>
+            <ButtonGroup fullWidth={true} variant="outlined">
+                <LoadingButton
+                    loading={awaitingConfirmation}
+                    disabled={publicKey == null || !message}
+                    onClick={async () => {
+                        if (publicKey == null) {
+                            return;
+                        }
+                        if (!supportedTransactionVersions) throw new Error("Wallet doesn't support versioned transactions!");
+                        if (!supportedTransactionVersions.has(0)) throw new Error("Wallet doesn't support v0 transactions!");
+
+                        const messageBuffer = new Uint8Array(message.split('').map((c) => c.charCodeAt(0)));
+
+                        /**
+                         * This lookup table only exists on devnet and can be replaced as
+                         * needed.  To create and manage a lookup table, use the `solana
+                         * address-lookup-table` commands.
+                         */
+                        const { value: lookupTable } = await connection.getAddressLookupTable(
```

Comment:
maybe didnt need the whole address lookup table thing to test V0 txns, but figured I would keep the code as is (stolen from `wallet-adapter`)